File: C++/StringSplicing/StringSplicing.cpp
#include <iostream>

std::string stringSplice(std::string word1, std::string word2) {
  std::string result;
  int mergedLength = (word1.length() > word2.length()) ? word1.length() : word2.length();

  for (int i = 0; i < mergedLength; i++) {
    if (i < word1.length()) {
      result += word1[i];
    }
    if (i < word2.length()) {
      result += word2[i];
    }
  }
  return result;
}

int main() {
  std::cout << stringSplice("Original", "Thought") << std::endl;
  return 0;
}

File: Rust/StringSplicing/stringSplicing.rs
fn spliceStrings(word1: &str, word2: &str) -> String {
  let mut result = "".to_string();
  let mergedLength = std::cmp::max(word1.len(), word2.len());

  for i in 0..mergedLength {
    if let Some(c) = word1.chars().nth(i) { result.push(c) };
    if let Some(c) = word2.chars().nth(i) { result.push(c) };
  }

  return result;
}

fn main() {
  println!("{}", spliceStrings("Basket", "Ball"));
}